src/socket.rs

```diff
@@ -27,12 +28,16 @@ pub struct UtpSocket<P> {
     cid_gen: Mutex<StdConnectionIdGenerator<P>>,
     accepts: mpsc::UnboundedSender<(Accept<P>, Option<ConnectionId<P>>)>,
     socket_events: mpsc::UnboundedSender<SocketEvent<P>>,
+    /// Maximum allowed connections, which only restricts outbound connections, but counts all
+    /// active connections.
+    max_conns: usize,
```

Comment:
what is the rationale on restricting only outbound connections?

Reply:
Hm, this one is a little fuzzy. I suppose that:
- outgoing connections feel delayable until we want to send the data, but inbound connections were presumably initiated by someone else, so we can't just wait as long as we want to accept
- we might have a preference to receive data (help selves) over sending it (help others), if we are resource constrained

Any thoughts on this? I suppose the alternative would be to add the same pausing mechanism into the accepts.

Reply:
I think the first point is more important than the second.

if we delay accepting a connection, then the initiating endpoint may give up on the connection before we accept, and then when we accept the connection we will consume a connection until the idle timeout.

uTP streams are bidirectional, and the accepting endpoint of a uTP stream is not necessarily the endpoint receiving data, so I don't think we can make assumptions about one endpoint being the sender and the other being the receiver.

that said, I think it's okay to only limit outgoing connections if we make it clear from the config that it is not total connections. if it becomes evident that an unlimited number of incoming (i.e. accepted) connections are harming outgoing connections, then we can address it with a different approach.